docs/resource/k8s.md

```diff
@@ -354,3 +354,70 @@ A CronJob creates Jobs on a repeating schedule.
 <!-- endsemconv -->
 
 [DocumentStatus]: https://opentelemetry.io/docs/specs/otel/document-status
+
+## Specify resource attributes using Kubernetes annotations
```

Comment:
I wonder if this should be an OTEP (or non-normative guide) instead. 
In the current format it looks more like guidance not a spec.

Also I wonder if that could be part of the `service.*` definition. What we are trying to do here is to define how `service.*` attributes should be populated in a specific environment not how K8s attributes are defined/affected.

@open-telemetry/specs-semconv-approvers thoughts?